Update some block dependencies to allow DataViews to work

louwie17 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This updates two dependencies with the blocks package to prevent the build process from using an older version of `@floating-ui/react-dom`. As this broke the use of dataviews in this PR: https://github.com/woocommerce/woocommerce/pull/56641

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure all the GH actions are green
2. Create a new post and change the editor to **Code Editor**
3. Paste `<!-- wp:woocommerce/handpicked-products {"editMode":false,"products":[]} /-->`
4. Select a product and change some of the settings, the block should work correctly
5. Now also remove the product afterwards, it should display a list of sample products and work correctly

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

louwie17 committed:
Update @wordpress/components dependency

louwie17 committed:
Update test snapshots

louwie17 committed:
Update the server-side-render dependency

louwie17 committed:
Add changelog

roykho reviewed (approved):
Great work! For my curiosity, we're using `wp-6.6` is because of `L-1`?

When specific product is selected:
![CleanShot 2025-03-25 at 12 17 19](https://github.com/user-attachments/assets/737c97d9-bfc0-46e1-8f7c-f583beb4c97c)

When product is removed:
![CleanShot 2025-03-25 at 12 17 05](https://github.com/user-attachments/assets/b65ac41e-c1e5-4d90-ae42-8fd89abbd9a0)

louwie17 commented:
> Great work! For my curiosity, we're using `wp-6.6` is because of `L-1`?

Good question, I believe `L-1` is the last two WP versions, so it is technically suppose to be 6.5, until 6.8 gets released. I updated it to 6.6 as this is what we have currently set in syncpack as the default for all the WP dependencies: https://github.com/woocommerce/woocommerce/blob/df495f62037815b8d7b8ea3ae6687058fbaaf877/.syncpackrc#L311

Most of the WP dependencies though are only used for development tasks, unit tests and types, given it uses the `window.wp` for most of them, which is the same case for the two I updated above.
